### ISSUE TITLE
can't open db's in hidden folders

### ISSUE THREAD

**Issue by Dan-Essin** [sqlitebrowser/sqlitebrowser#1604]
Details for the issue
--------------------

#### What did you do?
clicked open database
my db is in /Users/essin/Library/Application Support/Firefox/Profiles/0hh3dgwz.default-1532458593717/cookies.sqlite

#### What did you expect to see?
Expected to see the library folder so I could open it

#### What did you see instead?
no hidden folders were ;isted

Useful extra information
-------------------------

The info below often helps, please fill it out if you're able to. :)

#### What operating system are you using?

- [ ] Windows: ( _version:_ ___ )
- [ ] Linux: ( _distro:_ ___ )
- [ X] Mac OS: ( _version:_ ___ )
- [ ] Other: ___

#### What is your DB4S version?

- [ ] 3.11.0-alpha*
- [ X] 3.10.1
- [ ] Other: ___

#### Did you also

- [ ] Try out the latest nightly version: https://github.com/sqlitebrowser/sqlitebrowser#nightly-builds
- [X ] Search for an existing similar issue: https://github.com/sqlitebrowser/sqlitebrowser/issues?utf8=%E2%9C%93&q=is%3Aissue%20

**Comment by MKleusberg** [comment]
Did you already try to press <kbd>Command</kbd>+<kbd>Shift</kbd>+<kbd>.</kbd> while in the file dialog? Got that from [here](https://www.lifewire.com/display-hidden-files-in-os-x-open-and-save-dialog-boxes-2260878) but have no idea if it's working :wink:

**Comment by chrisjlocke** [comment]
@Dan-Essin *cough*

**Comment by sproctor** [comment]
I think I'm having a related issue. On linux, trying to open a database located in a hidden folder fails. In the file browser, the files do not appear (after choosing to show hidden files, then navigating to the hidden directory). Opening the file from the command line fails with `QIODevice::read (QFile, "prefs.db"): device not open`. If I move the same file to a non-hidden directory it opens fine.

**Comment by edav-mx** [comment]
Have the same behavior on linux, if path is in a "hidden" folder, the won't open, moving the same file to a non hidden path resolves the issue.

**Comment by kreegahbundolo** [comment]
This is still a problem in version 3.13.99 on Linux (Ubuntu 24.04). It's a bit weirder than just not showing folders though:

Right clicking in file dialog gives a dropdown menu where you can select to show hidden folders. Navigating to a hidden folder shows no files or folders inside it ...UNLESS you have opened a file inside there before (by drag/dropping it from os file explorer for example). This is very strange.

To sum up, if I have opened $HOME/.config/something/db.sqlite before I will be able to navigate to it using the DB browser file dialog. But all other folders and files inside $HOME/.config/ and $HOME/.config/something/ are invisible

**Comment by tikuma-lsuhsc** [comment]
I too am experiencing exactly what @kreegahbundolo described with the latest Snap versions (latest/stable 3.13.0-4078 as well as latest/edge 3.13.1-4209) on Ubuntu 25.04. Here are a couple screenshots:

Open dialog in the database folder:
<img width="1508" height="1146" alt="Image" src="https://github.com/user-attachments/assets/84660ee2-8fd4-4b31-bf4f-e7449fede264" />

Open dialog as I typed in the database file name:
<img width="1508" height="1146" alt="Image" src="https://github.com/user-attachments/assets/05d5f4ae-e473-450d-b47a-83675613628f" />

When I press OK, this error dialog pops up:
<img width="744" height="498" alt="Image" src="https://github.com/user-attachments/assets/8432d1ff-4656-4609-9201-66a911145a27" />

This database file opens normally by other viewers and can be opened if I move the database to elsewhere.
